subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c

```diff
@@ -560,21 +563,26 @@ int nrf_provisioning_req(void)
 		if (ret == -EINVAL) {
 			__ASSERT(false, "Invalid exchange, abort");
 			LOG_ERR("Invalid exchange");
+			dm.cb(NRF_PROVISIONING_EVENT_STOP, dm.user_data);
 		} else if (ret == -ECONNREFUSED) {
 			LOG_ERR("Connection refused, client exits");
 			k_mutex_lock(&np_mtx, K_FOREVER);
 			initialized = false;
 			k_mutex_unlock(&np_mtx);
+			dm.cb(NRF_PROVISIONING_EVENT_STOP, dm.user_data);
 			break;
 		} else if (ret < 0) {
 			LOG_ERR("Provisioning failed, error: %d", ret);
+			dm.cb(NRF_PROVISIONING_EVENT_STOP, dm.user_data);
 		} else if (ret > 0) {
 			/* Provisioning finished */
 			if (IS_ENABLED(CONFIG_NRF_PROVISIONING_SAVE_CMD_ID)) {
 				LOG_DBG("Saving the latest command id");
 				commit_latest_cmd_id();
 			}
-			dm.cb(dm.user_data);
+			dm.cb(NRF_PROVISIONING_EVENT_DONE, dm.user_data);
+		} else {
+			dm.cb(NRF_PROVISIONING_EVENT_STOP, dm.user_data);
 		}
 
```

Comment:
This is still not what we agreed.. STOP must follow the START unconditionally. Just to indicate that network activities have stopped.

So it should be
```c
		dm.cb(NRF_PROVISIONING_EVENT_START, dm.user_data);
		if (IS_ENABLED(CONFIG_NRF_PROVISIONING_HTTP)) {
			ret = nrf_provisioning_http_req(&rest_ctx);
		} else {
			ret = nrf_provisioning_coap_req(&coap_ctx);
		}
		dm.cb(NRF_PROVISIONING_EVENT_STOP, dm.user_data);
```

Then `dm.cb(NRF_PROVISIONING_EVENT_DONE, dm.user_data);` should only come if there was provisioning.

Reply:
Oops, sorry my bad. Fixed the events.